client/rpc-servers/src/lib.rs

```diff
@@ -104,9 +109,14 @@ pub fn start_ws<M: Send + Sync + 'static>(
 	let mut builder = WsServerBuilder::new()
 		.max_request_body_size(max_request_body_size as u32)
 		.max_connections(max_connections as u64)
-		.set_id_provider(RandomStringIdProvider::new(16))
 		.custom_tokio_runtime(rt.clone());
 
+	if let Some(provider) = id_provider {
+		builder = builder.set_id_provider(provider);
+	} else {
+		builder = builder.set_id_provider(RandomStringIdProvider::new(16));
+	};
```

Comment:
So strings is the default? Hmm, I'd have liked to have numbers be the default and let users that need it switch. Thoughts?

Reply:
I agree but it's to make it backward compatible with the current master.

On our `jsonrpsee node` some clients didn't manage to parse subscription IDs as number, most likely some JS library but polkadot-js works.

Thus, this is to be on the safe side.

Reply:
I see. Ok, fair enough.